pkg/controller/core/localqueue_controller.go

```diff
@@ -146,20 +157,45 @@ func (r *LocalQueueReconciler) Delete(e event.DeleteEvent) bool {
 }
 
 func (r *LocalQueueReconciler) Update(e event.UpdateEvent) bool {
-	q, match := e.ObjectNew.(*kueue.LocalQueue)
-	if !match {
+	oldLq, oldIsLq := e.ObjectOld.(*kueue.LocalQueue)
+	newLq, newIsLq := e.ObjectNew.(*kueue.LocalQueue)
+	if !oldIsLq || !newIsLq {
 		// No need to interact with the queue manager for other objects.
 		return true
 	}
-	log := r.log.WithValues("localQueue", klog.KObj(q))
+	log := r.log.WithValues("localQueue", klog.KObj(newLq))
 	log.V(2).Info("Queue update event")
-	if err := r.queues.UpdateLocalQueue(q); err != nil {
-		log.Error(err, "Failed to update queue in the queueing system")
+
+	oldStopPolicy := ptr.Deref(oldLq.Spec.StopPolicy, kueue.None)
+	newStopPolicy := ptr.Deref(newLq.Spec.StopPolicy, kueue.None)
+
```

Comment:
maybe something like 
```go
if newStopPolicy==oldStopPolicy {
  if  newStopPolicy == None {
     // update in cache/queue
  }
  return true
} 

if  newStopPolicy == None {
     // add to cache/queue
} else {
     // rm from cache/queue
}

```

is easier to follow

Reply:
Done